Date: Nov 20 17:56:57 2018

From: cc115ed5b0db830f7c6bc73081825207084ff3e0 Dec 17 21:58:08 2018

Subject: [PATCH] DELIA-30340 : Ignore pkill request with empty pattern

        Suspecting some race condition from DS module is initiating pkill with empty pattern
        This could be probable cause for systemwide shutdown after deepsleep wakeup.

Source: COMCAST

Upstream-Status: Pending

Signed-off-by: skv01 <Shibu_KV@cable.comcast.com>
---

diff --git a/pgrep.c b/pgrep.c
index 539a2d8..461dd89 100644
--- a/pgrep.c
+++ b/pgrep.c
@@ -716,6 +716,8 @@ static void parse_opts (int argc, char **argv)
 	if (strstr (program_invocation_short_name, "pkill")) {
 		int sig;
 		i_am_pkill = 1;
+                // DELIA-30340   - Temporarily enabling debug to identify possible candidate issuing systemwide kill after deepsleep wakeup 
+                opt_echo = 1 ;
 		sig = signal_option(&argc, argv);
 		if (-1 < sig)
 			opt_signal = sig;
@@ -895,6 +897,18 @@ static void parse_opts (int argc, char **argv)
 		xerrx(EXIT_USAGE, _("no matching criteria specified\n"
 				     "Try `%s --help' for more information."),
 				     program_invocation_short_name);
+
+        // Clean up all preceding spaces before comparison 
+
+        if ( 1 == i_am_pkill ) {
+                while(isspace(*opt_pattern)) opt_pattern++ ;
+                if ( 0 == strncmp(opt_pattern, "", 1) ) {
+                    printf("pkill invoked from process with pid :  %d\n",(int) getppid());
+		    xerrx(EXIT_USAGE, _("matching criteria cannot be specified as empty\n"
+				     "Try `%s --help' for more information."),
+				     program_invocation_short_name);
+                }
+        }
 }
 
 
